src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/prisonperson/profiledetails/ProfileDetailsService.kt

```diff
@@ -28,11 +28,10 @@ class ProfileDetailsService(
           endDateTime = booking.getReleaseTimer(),
           latestBooking = booking.bookingSequence == 1,
           profileDetails = booking.profiles.first().profileDetails
-            .filter { pd -> pd.profileCodeId != null }
```

Comment:
Records with null profile codes get created for new offenders but also exist where the user has cleared the profile code. Don't filter them out here, that's up to the caller of the API.